article.md

```diff
@@ -0,0 +1,612 @@
+# So you want to use types with Sequelize?
+
+Once a team first takes the leap into the world of types with Typescript, it seems there's no going back. Typescript has become common to see in front-end applications, so why aren't we using all those juicy compile-time type checks on the server? At Vivacity, we've adopted Typescript on *both* the front- and back-end for new projects. We're currently building a new data set management tool using Typescript, Express and Sequelize. This guide will step you through how to set this up so you can create, read, update and delete without ever getting a type wrong.
+
+---
+
+## Plan
+
+1. Explain the website and schema we are going to make
+2. Setup: First, set up a default project with simple stuff.
+3. Api design: Set up project to use controllers.
+4. talk about how we're going to have each model have its own file. Except, rather than making it a default export, we're going to export a ModelFactory function, and two interfaces, ModelAttributes and ModelInstance. We avoid default exports because it makes type inference worse in typescript.
+5. Model loader: We then have an index.ts file, similar to with js, that loads them up. Except, we also need a DbInterface type. It does this and doesn't load all files in a directory, because otherwise you'd lose type inference
+6. Model definitions: Explain ModelAttrs, ModelInstance, SequelizeAttrs
+7. Associations and the mixins
+8. Glorious final result: writing api routes with lots of beautiful type inference (screenshot of intellisense)
+
+### Plan:  Schema
+
+- Users: firstName, lastName
+- Posts: AuthorId, title, text, category: enum("tech", "croissants", "techno")
+- Comments: PostId, AuthorId, text
+- PostUpvotes: PostId, UpvoterId
+
+Users (Author) 1:m Posts  
+Users (Author) 1:m Comments  
+Posts 1:m Comments  
+User (Upvoter) n:m Comments
+
+---
+
+## Let's get started
+
+We're about to embark on a rollercoaster adventure. At first, adding Typescript will seem like nothing but a cumbersome waste of time. Hours lost to just littering your Sequelize models with a slew of verbose type annotations. But, peservere! By the time we reach the end, your models will be so well typed that your api routes will practically be writing themselves. Here's a sneak peak.
+
+![image TODO][todo]
+
+Now you're enticed, let's get started. We will be creating a simple JSON api for a blog post website. Apparently, there's a big gap in the market for a site where tech nerds, croissant enthusiasts and techno heads can come together to post blogs. So, our website will allow blogs in these three different categories. Here's the database schema.
+
+> TODO: Probably, in future, change this to a schema picture like the one Martin made for persp.
```

Comment:
I'd be very happy to make a schema picture after demo :smiley:

Reply:
Thank you so much for doing that!